File: app/src/main/java/me/zhanghai/android/files/provider/content/ContentFileAttributes.java
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.provider.content;

import android.net.Uri;
import android.os.Parcel;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import me.zhanghai.android.files.provider.common.ParcelableContentProviderFileAttributes;

public class ContentFileAttributes extends ParcelableContentProviderFileAttributes {

    ContentFileAttributes(@Nullable String mimeType, long size, @NonNull Uri uri) {
        super(new ContentFileAttributesImpl(mimeType, size, uri));
    }


    public static final Creator<ContentFileAttributes> CREATOR =
            new Creator<ContentFileAttributes>() {
                @Override
                public ContentFileAttributes createFromParcel(Parcel source) {
                    return new ContentFileAttributes(source);
                }
                @Override
                public ContentFileAttributes[] newArray(int size) {
                    return new ContentFileAttributes[size];
                }
            };

    protected ContentFileAttributes(Parcel in) {
        super(in);
    }
}

File: app/src/main/java/me/zhanghai/android/files/provider/content/ContentFileAttributes.kt
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.provider.content

import android.net.Uri
import android.os.Parcel
import android.os.Parcelable
import me.zhanghai.android.files.provider.common.ParcelableContentProviderFileAttributes

internal class ContentFileAttributes : ParcelableContentProviderFileAttributes {
    constructor(mimeType: String?, size: Long, uri: Uri) : super(
        ContentFileAttributesImpl(mimeType, size, uri)
    )

    private constructor(source: Parcel) : super(source)

    companion object {
        @JvmField
        val CREATOR = object : Parcelable.Creator<ContentFileAttributes> {
            override fun createFromParcel(source: Parcel): ContentFileAttributes =
                ContentFileAttributes(source)

            override fun newArray(size: Int): Array<ContentFileAttributes?> = arrayOfNulls(size)
        }
    }
}
